node/src/neighborhood/mod.rs

```diff
@@ -3064,6 +3148,267 @@ mod tests {
+        subject.logger = Logger::new(test_name);
+        let subject_addr = subject.start();
+        let peer_actors = peer_actors_builder().ui_gateway(ui_gateway).build();
+        let assertion_msg = AssertionsMessage {
+            assertions: Box::new(move |neighborhood: &mut Neighborhood| {
+                assert_eq!(
+                    neighborhood.exit_locations_opt,
+                    Some(HashMap::from([
+                        (1usize, vec!["CZ".to_string(), "SK".to_string()]),
+                        (2usize, vec!["AT".to_string(), "DE".to_string()]),
+                        (3usize, vec!["PL".to_string(), "HU".to_string()]),
+                    ]))
+                );
+                assert_eq!(neighborhood.fallback_routing, true);
+            }),
+        };
+        subject_addr.try_send(BindMessage { peer_actors }).unwrap();
+
+        subject_addr.try_send(message).unwrap();
+        subject_addr.try_send(assertion_msg).unwrap();
+
+        System::current().stop();
+        system.run();
+
+        TestLogHandler::new().assert_logs_contain_in_order(vec![
+            &format!("INFO: {}: Fallback Routing is Set.", test_name),
+            &format!("INFO: {}: Exit Location Set:", test_name),
+        ]);
+        TestLogHandler::new().assert_logs_contain_in_order(vec![
+            &format!("INFO: {}: Exit Location Set:", test_name),
+            &format!("{}", "Country Codes: [\"CZ\", \"SK\"] - Priority: 1;"),
+        ]);
+        TestLogHandler::new().assert_logs_contain_in_order(vec![
+            &format!("INFO: {}: Exit Location Set:", test_name),
+            &format!("{}", "Country Codes: [\"AT\", \"DE\"] - Priority: 2;"),
+        ]);
+        TestLogHandler::new().assert_logs_contain_in_order(vec![
+            &format!("INFO: {}: Exit Location Set:", test_name),
+            &format!("{}", "Country Codes: [\"PL\", \"HU\"] - Priority: 3;"),
+        ]);
+    }
+
+    #[test]
+    fn exit_location_is_set_and_unset_with_fallback_routing_using_exit_location_msg() {
+        init_test_logging();
+        let test_name =
+            "exit_location_is_set_and_unset_with_fallback_routing_using_exit_location_msg";
+        let request = UiSetExitLocationRequest {
+            fallback_routing: false,
+            exit_locations: vec![
+                CountryCodes {
+                    country_codes: vec!["CZ".to_string()],
+                    priority: 1,
+                },
+                CountryCodes {
+                    country_codes: vec!["FR".to_string()],
+                    priority: 2,
+                },
+            ],
+        };
+        let message = NodeFromUiMessage {
+            client_id: 0,
+            body: request.tmb(0),
+        };
+        let mut subject = make_standard_subject();
+        let system = System::new(test_name);
+        let (ui_gateway, _, _) = make_recorder();
+
+        subject.exit_locations_opt = None;
+        subject.logger = Logger::new(test_name);
+
```

Comment:
These two blank lines make this look like the Act portion of the test. It's obvious upon further examination that it's not, but they're a little distracting.